nuxeo-features/nuxeo-automation/nuxeo-automation-features/src/main/java/org/nuxeo/ecm/automation/core/operations/services/bulk/BulkRunAction.java

```diff
@@ -103,16 +106,34 @@ public BulkStatus run() throws IOException {
             throw new NuxeoException("Action '" + action + "' denied", HttpServletResponse.SC_FORBIDDEN);
         }
 
+        if (query == null && providerName == null) {
+            throw new OperationException("Query and ProviderName cannot be both null");
+        }
+
         String userName = session.getPrincipal().getName();
 
         PageProviderDefinition def = query != null ? PageProviderHelper.getQueryPageProviderDefinition(query)
                 : PageProviderHelper.getPageProviderDefinition(providerName);
-        PageProvider provider = PageProviderHelper.getPageProvider(session, def, namedParameters,
+
+        if (def == null) {
+            throw new OperationException("Could not get Provider Definition from either query or provider name");
+        }
+
+        PageProvider<?> provider = PageProviderHelper.getPageProvider(session, def, namedParameters,
                 queryParams != null ? queryParams.toArray(new String[0]) : null);
         query = PageProviderHelper.buildQueryString(provider);
 
-        Map<String, Serializable> mapParams = BulkParameters.paramsToMap(parametersAsJson);
-        BulkCommand.Builder builder = new BulkCommand.Builder(action, query).user(userName).params(mapParams);
+        if (query.contains("?")) {
+            throw new OperationException("Query parameters could not be parsed");
+        }
+
+        BulkCommand.Builder builder = new BulkCommand.Builder(action, query).user(userName);
+        try {
+            Map<String, Serializable> mapParams = BulkParameters.paramsToMap(parametersAsJson);
+            builder.params(mapParams);
+        } catch (JsonParseException e) {
```

Comment:
You should catch `IOException` as it's the exception in method signature + we'll probably catch more stuff.